### PR TITLE
Add switch component

### PR DESCRIPTION
**Pull request by seedy**
## Issue

#145 

## How to test

- Run storybook
- Check story : Switch

## Design opinion required on
- Disabled state : not part of mockups
- Focus state: not part of mockups. Should we prefer an elevation ?
- Hovered state: not part of mockups. Should we prefer an elevation ?
- Sizes: already existing and updated without mockups

Rest follows design preco from the issue

blocked by https://github.com/traefik/faency/pull/164

### CODE REVIEW

**Comment by seedy** [comment]
@matthieuh I updated the component theme with global theme changes and simplified the thumb behaviour : when switch is focused or hovered, I apply some elevation variant.

WDYT about the elevation behaviour ?

File: components/Switch/Switch.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+          height: THUMB_DIAMETER * LARGE_RATIO,
+          transform: `translateX(${LARGE_RATIO * THUMB_GAP})`,
+
+          '&[data-state="checked"]': {
+            transform: `translateX(${LARGE_RATIO * (ROOT_WIDTH - THUMB_GAP - THUMB_DIAMETER)}px)`,
+          },
+        },
+      },
+    },
+    disabled: {
```
Comment (matthieuh) [inline components/Switch/Switch.tsx:92]: When a Switch is disabled element should not be focusable to be consistent with buttons and inputs.

File: components/Switch/Switch.themes.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import tinycolor from 'tinycolor2';
+import { Property } from '@stitches/react/types/css';
+export namespace Theme {
+  type Colors = {
```
Comment (matthieuh) [inline components/Switch/Switch.themes.ts:4]: type is missing `switchActiveBackground`

**Review by matthieuh — CHANGES_REQUESTED** [review]
I added some request changes but otherwise great work! 🎉

File: components/Switch/Switch.tsx (hunk 1 above)
Comment (seedy) [reply components/Switch/Switch.tsx]: fixed, I replaced the variant by `&:disabled`

File: components/Switch/Switch.themes.ts (hunk 2 above)
Comment (seedy) [reply components/Switch/Switch.themes.ts:6]: fixed, I also updated from upstram

**Review by matthieuh — APPROVED** [review]
LGTM

**Review by paulocfjunior — APPROVED** [review]
LGTM, friendly reminder to export the `Switch` component from the root `index.ts`, otherwise, we won't be able to import it on another app

**Pull request merged**